pkg/api/handlers/compat/images_search.go

```diff
@@ -77,5 +78,18 @@ func SearchImages(w http.ResponseWriter, r *http.Request) {
 		utils.BadRequest(w, "term", query.Term, err)
 		return
 	}
-	utils.WriteResponse(w, http.StatusOK, results)
+
+	compatResults := []registry.SearchResult{}
```

Comment:
```suggestion
	compatResults := make([]registry.SearchResult,0,len(results))
```
You should preallocate memory for the slice to reduce the memory copy operations.